ibis/tests/all/test_geospatial.py

```diff
@@ -1,48 +1,83 @@
 """ Tests for geo spatial data types"""
-from inspect import isfunction
+import sys
 
 import numpy as np
 import pytest
 from numpy import testing
+from pytest import param
 
 import ibis
-from ibis.tests.backends import MapD
+from ibis.tests.backends import MapD, PostgreSQL
+
+GEODATAFRAME_SUPPORTED = False
+
+try:
+    # supported just for python >= 3.6
+    if sys.version_info >= (3, 6):
+        import geopandas
+        import shapely.wkt  # noqa: F401
+
+        GEODATAFRAME_SUPPORTED = True
+except ImportError:
+    ...
+
 
 # geo literals declaration
-point_0 = ibis.literal((0, 0), type='point:geometry').name('p')
-point_1 = ibis.literal((1, 1), type='point:geometry').name('p')
-point_2 = ibis.literal((2, 2), type='point;4326:geometry').name('p')
-point_3 = ibis.literal((1, 1), type='point:geography').name('p')
-point_4 = ibis.literal((2, 2), type='point;4326:geography').name('p')
+point_geom_0 = ibis.literal((0, 0), type='point;4326:geometry').name('p')
+point_geom_1 = ibis.literal((1, 1), type='point;4326:geometry').name('p')
+point_geom_2 = ibis.literal((2, 2), type='point;4326:geometry').name('p')
+point_geog_0 = ibis.literal((0, 0), type='point;4326:geography').name('p')
+point_geog_1 = ibis.literal((1, 1), type='point;4326:geography').name('p')
+point_geog_2 = ibis.literal((2, 2), type='point;4326:geography').name('p')
 polygon_0 = ibis.literal(
     (
         ((1, 0), (0, 1), (-1, 0), (0, -1), (1, 0)),
         ((0.1, 0), (0, 0.1), (-0.1, 0), (0, -0.1), (0.1, 0)),
     ),
-    type='polygon',
+    type='polygon;4326:geometry',
 )
 
 # add here backends that support geo spatial types
 all_db_geo_supported = [MapD]
+if GEODATAFRAME_SUPPORTED:
+    all_db_geo_supported.append(PostgreSQL)
 
 
 @pytest.mark.parametrize(
     ('expr_fn', 'expected'),
     [
-        (lambda t: t['geo_linestring'].length(), [1.41] * 5),
-        (lambda t: t['geo_polygon'].perimeter(), [5.66] * 5),
-        (lambda t: t['geo_point'].x(), [0, 1, 2, 3, 4]),
-        (lambda t: t['geo_point'].y(), [0, 1, 2, 3, 4]),
-        (lambda t: t['geo_linestring'].x_min(), [0, 1, 2, 3, 4]),
-        (lambda t: t['geo_linestring'].x_max(), [1, 2, 3, 4, 5]),
-        (lambda t: t['geo_linestring'].y_min(), [0, 1, 2, 3, 4]),
-        (lambda t: t['geo_linestring'].y_max(), [1, 2, 3, 4, 5]),
-        (lambda t: t['geo_multipolygon'].n_points(), [12] * 5),
-        (lambda t: t['geo_multipolygon'].n_rings(), [4] * 5),
-        (lambda t: t['geo_point'].srid(), [0] * 5),
+        param(lambda t: t['geo_linestring'].length(), [1.41] * 5, id='length'),
+        param(lambda t: t['geo_point'].x(), [0, 1, 2, 3, 4], id='x'),
+        param(lambda t: t['geo_point'].y(), [0, 1, 2, 3, 4], id='y'),
+        param(
+            lambda t: t['geo_linestring'].x_min(),
+            [0, 1, 2, 3, 4],
+            id='x_min'),
```

Comment:
This formatting is not consistent with black. Can you make sure to run `pre-commit run black --all-files` to ensure that formatting is run. I'm not sure why this wouldn't get picked up when you commit.

Reply:
maybe it happened bc I was using some old version of black .. I recreated my environment .. so maybe it will work properly now. I will run this command here. thanks!

Reply:
I ran that .. but it seems nothing changed ...